### PR TITLE
Remove pytest warnings

### LINKED ISSUE #5300: Remove all warnings from pytest
When running `tox` we see these warnings in the summary.

We should use `request` fixture and access to `request.config` instead.

Docs: https://docs.pytest.org/en/latest/fixture.html#request-context
Change log: https://docs.pytest.org/en/latest/deprecations.html#pytest-config-global

```
====================================================================================== warnings summary ======================================================================================
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_mkdocs
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_mkdocs_index
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_mkdocs_index_no_directory_urls
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_mkdocs_no_directory_urls
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_and_version
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_and_version_and_page
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_and_version_and_page_htmldir
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_and_version_and_page_signlehtml
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_and_version_htmldir
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_and_version_singlehtml
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_only
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_only_htmldir
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_project_only_singlehtml
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_restructured_text
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_restructured_text_invalid
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_and_version
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_and_version_and_page
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_and_version_and_page_htmldir
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_and_version_and_page_singlehtml
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_and_version_htmldir
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_and_version_singlehtml
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_only
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_only_htmldir
readthedocs/rtd_tests/tests/test_core_tags.py::CoreTagsTests::test_translation_project_only_singlehtml
  /home/humitos/rtfd/code/readthedocs-corporate/.tox/py36/readthedocs.org/readthedocs/rtd_tests/tests/test_core_tags.py:19: PytestDeprecationWarning: the `pytest.config` global is deprecated.  Please use `request.config` or `pytest_configure` (if you're a pytest plugin) instead.
    scheme=pytest.config.option.url_scheme,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

### PR DESCRIPTION
**Pull request by saadmk11**
As i wasn't able to pass the request Fixture to the `setUp()` method I have Done it Like this Please Let Me know if There is a better implementation.
closes #5300

### CODE REVIEW

File: readthedocs/rtd_tests/tests/test_core_tags.py [hunk 1]
```diff
@@ -14,9 +14,13 @@
 class CoreTagsTests(TestCase):
     fixtures = ['eric', 'test_data']
 
+    @pytest.fixture(autouse=True)
```
Comment (safwanrahman) [inline readthedocs/rtd_tests/tests/test_core_tags.py]: I dont think fixture works in unittest style tests. Can you remove the fixture?

Comment (saadmk11) [reply readthedocs/rtd_tests/tests/test_core_tags.py]: Yes It Doesn't Work on the unittest Style Tests but to remove the pytest warnings we have to use the request fixture to access `config.option.url_scheme ` as The `pytest.config` object is deprecated. Is there any better way to get the same result?

File: conftest.py [hunk 2]
```diff
@@ -47,6 +47,12 @@ def pytest_configure(config):
...
 
+
 @pytest.fixture
 def api_client():
     return APIClient()
+
+
+@pytest.fixture(scope="class")
+def url_scheme(request):
+        request.cls.url_scheme = request.config.option.url_scheme
```
Comment (safwanrahman) [inline conftest.py]: use 4 space instead of tab

Comment (saadmk11) [reply conftest.py]: sorry for that. fixed!

**Comment by stsewd** [comment]
Thanks

**Pull request merged**